---
title: "Untitled"
output: html_document
date: "2024-10-17"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Starting point is to give them a couple of sequences from an Influenza virus. The idea of the first part is to identify which genus of Influenza they are from and whether they are related (come from the same outbreak) or are two independent infections. 

```{r}
library(tidyverse); library(rentrez); library(phylotools)
seqs <- entrez_fetch(db="nucleotide", id=c("PQ106526", "PQ032835", "PQ106516", "PQ106521"), rettype="fasta")
write(seqs, file="../data/Patientseqs.fasta")

```


Then, we want to know which type of Influenza virus is it. For this, they will be given a sampling 

## Data generation

```{r}
library(tidyverse); library(rentrez); library(phylotools)
metadata <- read.csv("../data/NCBImetadata.tsv", sep="\t")
selected <- metadata %>% filter(Protein == "hemagglutinin") %>% group_by(Species) %>% slice_sample(n=50) 
accessions <- selected %>% pull(Nucleotide)
seqs <- entrez_fetch(db="nucleotide", id=accessions, rettype="fasta")
write(seqs, file="../data/SpeciesSubsample.fasta")
fasta <- read.fasta("../data/SpeciesSubsample.fasta")
fasta$seq.name <- gsub(" .*", "", fasta$seq.name)
dat2fasta(fasta,"../data/SpeciesSubsample.fasta")
```


```{r}
library(ggtree)
tree <- read.tree("../data/Tree1.tree")

ggtree(tree, layout="daylight") %<+% metadata  + geom_tippoint(aes(color=Species)) 

```

```{r}
selected <- metadata %>% filter(Species == "Alphainfluenzavirus influenzae") %>% filter(Protein == "hemagglutinin") %>% group_by(Genotype) %>% slice_sample(n=10) 
accessions <- selected %>% pull(Accession)

seqs <- entrez_fetch(db="protein", id=accessions, rettype="fasta")
write(seqs, file="../data/InfASubsample.fasta")
```

